builder/hyperv/common/config.go

```diff
@@ -148,6 +148,21 @@ type CommonConfig struct {
 	// built. When this value is set to true, the machine will start without a
 	// console.
 	Headless bool `mapstructure:"headless" required:"false"`
+	// When configured, determines the device or device type that is given preferential
+	// treatment when choosing a boot device.
+	//
+	// For Generation 1:
+	//   - `IDE`
+	//   - `CD` *or* `DVD`
+	//   - `Floppy`
+	//   - `NET`
+	//
+	// For Generation 2:
+	//   - `IDE:x:y`
+	//   - `SCSI:x:y`
+	//   - `CD` *or* `DVD`
+	//   - `NET`
+	FirstBootDevice string `mapstructure:"first_boot_device" required:"false"`
 }
 
 func (c *CommonConfig) Prepare(ctx *interpolate.Context, pc *common.PackerConfig) ([]error, []string) {
```

Comment:
It would be nice to move some of the basic "does this input even make sense" validation you're doing in step_set_first_boot_device here in the config Prepare instead so that if it doesn't make sense `packer validate` will catch it and throw an error.

Reply:
This makes sense, it didn't occur to me to use the Prepare function.

Reply:
Sanity checks are now performed in the Prepare function and forces an early failure where appropriate.